### ISSUE TITLE
Why The development process often needs to be restarted

### ISSUE THREAD

**Issue by rccoder** [eggjs/egg#495]
* **Node Version**: 7.4.0
* **Egg Version**: 0.9.0
* **Plugin Name**: None
* **Plugin Version**: None
* **Platform**: macOS

Hi, I am using the egg framework and it is wonderful.

However, I have some question in my development.

if I change the code in development, always see the log:

```
2017-03-02 17:48:55,483 WARN 76489 [master] App Worker#1:76491 started at 7001, remain 0 (777ms)
2017-03-02 17:48:55,484 INFO 76489 [master] egg started on http://127.0.0.1:7001 (1158ms)
2017-03-02 17:50:34,876 WARN 76490 [agent:development] reload worker because /Users/rccoder/WeHIT/server/config/config.default.js___jb_old___ rename
2017-03-02 17:50:34,878 INFO 76489 [master] reload workers...
2017-03-02 17:50:34,887 INFO 76489 [master] App Worker#2:76543 start, state: none, current workers: ["1","2"]
2017-03-02 17:50:35,767 WARN 76489 [master] App Worker#2:76543 started at 7001, remain 0 (101061ms)
2017-03-02 17:50:35,871 WARN 76491 [app_worker] App Worker exit with signal SIGTERM
[Thu Mar 02 2017 17:50:35 GMT+0800 (CST)] [cfork:master:76489] worker:76491 disconnect (suicide: true, state: disconnected, isDead: false)
[Thu Mar 02 2017 17:50:35 GMT+0800 (CST)] [cfork:master:76489] don't fork new work (refork: false)
2017-03-02 17:50:35,890 INFO 76489 [master] App Worker#1:76491 disconnect, suicide: true, state: disconnected, current workers: ["2"]
(node:76489) DeprecationWarning: worker.suicide is deprecated. Please use worker.exitedAfterDisconnect.
[Thu Mar 02 2017 17:50:35 GMT+0800 (CST)] [cfork:master:76489] worker:76491 exit (code: 0, suicide: true, state: dead, isDead: true, isExpected: true)
```

Yeah, the project would not restart and do right, it would run the old version code.

So, I should input `Ctrl+C` to exit the project and then input `npm run dev` to start the project.

I do not know if it is my problem to let this case arise and how to avoid this case.

Thanks!

**Comment by popomore** [comment]
it's by design, if you change code, the worker will be restarted for reloading the new code.

**Comment by atian25** [comment]
> it would run the old version code.

@popomore @shaoshuai0102

**Comment by shaoshuai0102** [comment]
Usually the new started process will run the new codes. Are u sure it's running the old? Pls double check.

And if so, what's the editor u r using?

**Comment by rccoder** [comment]
Uh, I found it caused by the:

```
switch() {
  case 'xx' :
    ...
    break;
  default: 
   break;
}
```

the case was go to default and `break`，So, Refrash the page would happend noting, like run the old version code 😂

Thanks, It is my stupid mistake 😢

**Comment by shaoshuai0102** [comment]
😂

**Comment by brucewar** [comment]
![image](https://cloud.githubusercontent.com/assets/4013475/26049309/1cb53c60-398d-11e7-80e0-8b076370d2d8.png)

* Windows 10
* Egg.js 1.2.1
* Node.js 6.9.2

1. init egg with type simple
2. `npm run dev`
3. change file `home.js`
4. reload `127.0.0.1:7001`, the request is pending

**Comment by okoala** [comment]
@brucewar  Does the console print out the error message?

**Comment by brucewar** [comment]
@okoala no log after refreshing the page

**Comment by okoala** [comment]
@brucewar  I can not reproduce your situation. Is it blocked by browser plugins or system software? Maybe you can share your code.

**Comment by atian25** [comment]
@brucewar  try `curl`, if not worked, upload your mini project to github, and raise a new issue.

**Comment by brucewar** [comment]
only init project with `egg-init --type simple`
![image](https://cloud.githubusercontent.com/assets/4013475/26057885/03e64a8c-39ae-11e7-9fc6-7258225cd32a.png)

**Comment by atian25** [comment]
@brucewar  add `console.log` at controller action, try `curl`.

**Comment by brucewar** [comment]
![image](https://cloud.githubusercontent.com/assets/4013475/26088142/ffa25b44-3a27-11e7-9efa-d26cac99d3ac.png)
